src/cayenne/api/v1/doc.clj

```diff
@@ -1,50 +1,239 @@
-            [clojure.data.json :as json]))
-
-(def docs
-  {:apiVersion 0.1
-   :swaggerVersion 1.2
-   :apis [{:path "/works"}
-          {:path "/funders"}
-          {:path "/publishers"}]
-   :info {:title "CrossRef Unified Resource API"
-          :contact "labs@crossref.org"
-          :license "CC0 1.0 Universal"
-          :licenseUrl "http://creativecommons.org/publicdomain/zero/1.0"}})
-
-(defn api-path [path]
-  {:path path
-   :operations [{:method :GET
-                 :responseMessages {:code 404
-                                    :message "Entity not found."}}]})
-
-(def works-docs
-  {:apiVersion 01
-   :swaggerVersion 1.2
-   :resourcePath "/works"
-   :apis [(api-path "/works")
-          (api-path "/works/{doi}")
-          (api-path "/works/random/{count}")]})
-
-(def funders-docs
-  {:apiVersion 01
-   :swaggerVersion 1.2
-   :resourcePath "/funders"
-   :apis [(api-path "/funders")
-          (api-path "/funders/{funderId}")
-          (api-path "/funders/{funderId}/works")]})
-
-(def publishers-docs
-  {:apiVersion 01
-   :swaggerVersion 1.2
-   :resourcePath "/publishers"
-   :apis [(api-path "/publishers")
-          (api-path "/publishers/{ownerPrefix}")
-          (api-path "/publishers/{ownerPrefix}/works")]})
+  (:require [cayenne.api.v1.schema :as sc]
+            [cayenne.api.v1.filter :refer [std-filters compound-fields]]
+            [cayenne.api.v1.fields :refer [work-fields]]
+            [cayenne.api.v1.facet :refer [std-facets]]
+            [cayenne.api.v1.query :refer [select-fields sort-fields]]
+            [compojure.core :refer [defroutes GET]]
+            [clojure.data.json :as json]
+            [clojure.java.io :refer [resource]]
+            [ring.swagger.swagger-ui :refer [swagger-ui]]
+            [ring.swagger.swagger2 :as rs]
+            [schema.core :as s]))
+
+(def info
+  {:info
+   {:version "0.1"
+    :title "Crossref Unified Resource API"
+    :description (slurp (resource "description.md"))
+    :termsOfService "https://github.com/CrossRef/rest-api-doc"
+    :contact {:name "Crossref"
+              :email "support@crossref.org"
+              :url "https://crossref.org"}}})
+
+(def tags
+  {:tags
+   [{:name "Funder"
+     :description "Endpoints that expose funder related data"}
+    {:name "Journal"
+     :description "Endpoints that expose journal related data"}
+    {:name "Work"
+     :description "Endpoints that expose works related data"}
+    {:name "Prefix"
+     :description "Endpoints that expose prefix related data"}
+    {:name "Member"
+     :description "Endpoints that expose member related data"}
+    {:name "Type"
+     :description "Endpoints that expose type related data"}]})
+
+(defn- fields [compound-fields field]
+  (let [c-fields (get compound-fields (keyword field))
+        field-prefix (str "\n  + " field ".")]
+    [field
+     (if c-fields
+       (str field "." (clojure.string/join field-prefix c-fields)))]))
```

Comment:
Is this missing an else clause?

Reply:
I don't think so, if there aren't any compound fields then we don't need to do anything. Should probably refactor that to a `when` actually, will do that.